resources/nspawn.go

```diff
@@ -0,0 +1,306 @@
+			"org.freedesktop.machine1.NoSuchMachine" {
+			return false, err
+		}
+		exists = false
+		// if we could not successfully get the properties because
+		// there's no such machine the machine is stopped
+		// error if we need the image ignore if we don't
+		_, err = conn.GetImage(obj.GetName())
+		if err != nil && obj.State != stopped {
+			return false, fmt.Errorf(
+				"No machine nor image named '%s'",
+				obj.GetName())
+		}
+	}
+
+	// if the machine doesn't exist and is supposed to be stopped or
+	// the state matches we're done
+	if !exists && obj.State == stopped || properties["State"] == obj.State {
+		return true, nil
+	}
+
+	if obj.State == running {
+		// start the machine using svc resource
+		var serviceName = fmt.Sprintf(nspawnServiceTmpl,
+			obj.GetName())
+		log.Printf("Start service: %s", serviceName)
+		obj.svc.Name = serviceName
+		obj.svc.State = running
+		return obj.svc.CheckApply(apply)
+
+	}
+	if obj.State == stopped {
+		// terminate the machine with
+		// org.freedesktop.machine1.Manager.KillMachine
+		log.Printf("Stopping nspawn machine: %s", obj.GetName())
+		if apply {
+			if err := conn.KillMachine(obj.GetName()); err != nil {
+				return false, err
+			}
+		}
+		return true, nil
+	}
+	return false, nil // if we didn't succeed we failed
+}
+
+// NspawnUID is a unique resource identifier
+type NspawnUID struct {
+	// NOTE: there is also a name variable in the BaseUID struct, this is
+	// information about where this UID came from, and is unrelated to the
+	// information about the resource we're matching. That data which is
+	// used in the IFF function, is what you see in the struct fields here
+	BaseUID
+	name string // the machine name
+}
+
+// IFF aka if and only if they are equivalent, return true. If not, false
+func (obj *NspawnUID) IFF(uid ResUID) bool {
+	res, ok := uid.(*NspawnUID)
+	if !ok {
+		return false
+	}
+	return obj.name == res.name
+}
+
+// GetUIDs includes all params to make a unique identification of this object
+// most resources only return one although some resources can return multiple
+func (obj *NspawnRes) GetUIDs() []ResUID {
+	x := &NspawnUID{
+		BaseUID: BaseUID{name: obj.GetName(), kind: obj.Kind()},
+		name:    obj.Name, // svc name
+	}
+	return []ResUID{x}
+}
+
+// GroupCmp returns whether two resources can be grouped together or not
+func (obj *NspawnRes) GroupCmp(r Res) bool {
+	_, ok := r.(*NspawnRes)
+	if !ok {
+		return false
+	}
+	// TODO: depending on if the systemd service api allows batching we
+	// might be able to build this, although not sure how useful it is
+	// it might just eliminate parallelism be bunching up the graph
+	return false // not possible atm
+}
+
+// Compare two resources and return if they are equivalent
+func (obj *NspawnRes) Compare(res Res) bool {
+	switch res.(type) {
+	case *NspawnRes:
+		res := res.(*NspawnRes)
+		if !obj.BaseRes.Compare(res) {
+			return false
+		}
+		if obj.Name != res.Name {
+			return false
+		}
```

Comment:
Because we're using a composite resource, I think we need to sub-call Compare on the obj.svc too. Sorry that none of this was documented anywhere, you're the first person writing a composite resource but we'll be able to look at you as a future model, so let's get it perfect right away! Hurrah :)